miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -123,6 +123,13 @@ public Boolean isMetrixEnabled() {
     Map<String, String> misoProperties = exporter.getResolvedProperties();
     return misoProperties.containsKey("miso.notification.interop.enabled") && Boolean.parseBoolean(misoProperties.get("miso.notification.interop.enabled"));
   }
+  
+  @ModelAttribute("autoGenerateIdBarcodes")
+  public Boolean autoGenerateIdentificationBarcodes() {
+    MisoPropertyExporter exporter = (MisoPropertyExporter)applicationContextProvider.getApplicationContext().getBean("propertyConfigurer");
```

Comment:
This causes a compiler warning because ApplicationContextProvider.getApplicationContext() is a static method.  Perhaps it shouldn't be static - but I have run the application with this as a static call and it seems to work fine.